Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/Detect_features_in_polyhedra.h

```diff
@@ -0,0 +1,329 @@
+// Copyright (c) 2010 INRIA Sophia-Antipolis (France).
+// All rights reserved.
+//
+// This file is part of CGAL (www.cgal.org).
+// You can redistribute it and/or modify it under the terms of the GNU
+// General Public License as published by the Free Software Foundation,
+// either version 3 of the License, or (at your option) any later version.
+//
+// Licensees holding a valid commercial license may use this file in
+// accordance with the commercial license agreement provided with the software.
+//
+// This file is provided AS IS with NO WARRANTY OF ANY KIND, INCLUDING THE
+// WARRANTY OF DESIGN, MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE.
+//
+// $URL$
+// $Id$
+//
+//
+// Author(s)     : Laurent Rineau, Stephane Tayeb
+//
+
+#ifndef CGAL_POLYGON_MESH_PROCESSING_DETECT_FEATURES_IN_POLYHEDRA_H
+#define CGAL_POLYGON_MESH_PROCESSING_DETECT_FEATURES_IN_POLYHEDRA_H
+
+#include <CGAL/license/Polygon_mesh_processing.h>
+
+
+#include <CGAL/Kernel/global_functions_3.h>
+#include <CGAL/Polygon_mesh_processing/Detect_features_in_polyhedra_fwd.h>
+#include <CGAL/Compare_handles_with_or_without_timestamps.h>
+#include <CGAL/Polygon_mesh_processing/compute_normal.h>
+#include <set>
+
+namespace CGAL {
+namespace Polygon_mesh_processing {
+
+  template <typename Polyhedron, typename FT, typename PatchId_pmap>
+void detect_features(Polyhedron& p,
```

Comment:
This function should be documented, and use named parameters. I will add some hints in the main discussion